Cargo.toml

```diff
@@ -30,16 +30,15 @@ cassowary = "0.3"
 compact_str = "0.7.1"
 crossterm = { version = "0.27", optional = true }
 document-features = { version = "0.2.7", optional = true }
-indoc = "2.0"
 itertools = "0.12"
 lru = "0.12.0"
 paste = "1.0.2"
 serde = { version = "1", optional = true, features = ["derive"] }
 stability = "0.2.0"
 strum = { version = "0.26", features = ["derive"] }
-time = { version = "0.3.11", optional = true, features = ["local-offset"] }
 termion = { version = "3.0", optional = true }
 termwiz = { version = "0.22.0", optional = true }
+time = { version = "0.3", optional = true, features = ["local-offset"] }
```

Comment:
https://docs.rs/time/0.3.9/time/struct.Date.html#method.replace_day was introduced in 0.3.9 and is used in the calendar widget. Not sure if there's a specific need for something from the 0.3.11 version, but the downgrading that dep is out of scope for this PR.

`cargo +nightly update -Zdirect-minimal-versions && cargo check --all-features --all-targets` suggests we actually need:
```
itertools 0.12.1
serde 1.0.166
```

Reply:
The main change was the sorting. Personally I don't like seeing the fix version here as it doesn't change anything for the version resolution. (`3` and `3.0.1` will use `3.4.5`). Either the one from the lock file or the latest one is used.

With Rust edition 2024 taking MSRV into account it might actually be useful but before that it isn't.

And as a lot of crates interpret SemVer differently it's a mess anyway… `0.3` indicates it's not stable yet anyway so that's another mess by itself…

This PR is not about the version ranges so maybe just add the pointless bugfix version to skip this discussion?

Reply:
8aa7c6c9e